lib/elixir/test/elixir/gen_server_test.exs

```diff
@@ -50,8 +50,20 @@ defmodule GenServerTest do
     assert GenServer.cast(:foo, {:push, :world}) == :ok
   end
 
+  @tag capture_log: true
   test "call/3 exit messages" do
+    {:ok, pid} = GenServer.start_link(Stack, [:hello])
+
+    {:ok, stoped_pid} = GenServer.start_link(Stack, [:hello])
+    GenServer.stop(stoped_pid)
+
+    {:ok, killed_pid} = GenServer.start(Stack, [:hello], name: :stack)
+    Process.exit(killed_pid, :kill) # Simulating condition when GenServer killed
+    
     assert catch_exit(GenServer.call(self, :pop, 5000)) == {:calling_self, {GenServer, :call, [self, :pop, 5000]}}
+    assert catch_exit(GenServer.call(pid, :pop, 0)) == {:timeout, {GenServer, :call, [pid, :pop, 0]}}
+    assert catch_exit(GenServer.call(stoped_pid, :pop, 5000)) == {:noproc, {GenServer, :call, [stoped_pid, :pop, 5000]}}
+    assert catch_exit(GenServer.call(killed_pid, :pop, 5000)) == {:killed, {GenServer, :call, [killed_pid, :pop, 5000]}}
```

Comment:
This test has a race condition here. We could get `:noproc` instead of `:killed` if the process has already exited before this call. I don't think we need to test `:killed`. Testing `:calling_self`, `:timeout` and `:noproc` should be enough.